app/src/main/java/org/oppia/android/app/onboarding/OnboardingActivity.kt

```diff
@@ -15,7 +15,6 @@ class OnboardingActivity : InjectableAppCompatActivity(), RouteToProfileListList
   companion object {
     fun createOnboardingActivity(context: Context): Intent {
       val intent = Intent(context, OnboardingActivity::class.java)
-      intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
```

Comment:
Correct me if I am wrong but I think this should not be removed.

Reply:
Ummmm yeah! I guess removing this will make a new instance of this activity again and again!
SO this should be not removed True! but if I do not remove this the test is failing, so I got to think of doing something else.
I think creating a new createOnboardingActivity_Test for label test will do the needful.
Any inputs to this? @MaskedCarrot

Reply:
i replacing this line with ` .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)` will be the solution to this problem.
adding @rt4914 here to verify that this is the correct way of doing this.

Reply:
Yea ig too this can be a solution, will try this, also check if it works and commit once @rt4914 confirms.

Reply:
Hmm this is confusing to me. @rt4914 why do we need to clear top for this activity in the first place? SplashActivity should be calling finish() so there shouldn't be anything to clear.

Similarly, starting a new task in the backstack doesn't seem correct here for the same reason.

Reply:
Actually even I was confused on this part. I am not sure why this was introduced in the first place. For current implementation I think we can remove it  completely.

Reply:
Should I remove this completely? addFlags

Reply:
Yes.